src/TrimNoOps.cpp

```diff
@@ -153,9 +153,19 @@ class IsNoOp : public IRVisitor {
              op->name == Call::image_store ||
              op->name == Call::copy_memory)) {
             condition = const_false();
-        } else {
-            IRVisitor::visit(op);
+            return;
+        } else if (((op->name == Call::predicated_load) ||
+                    (op->name == Call::predicated_store))) {
+            // If the predicate of a predicated load/store is always false,
```

Comment:
I think it would be better if the simplifier handled the case where the predicate is always false.